test/lib/yggdrasil/player/registry_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule PlayerRegistryTests do
+  use ExUnit.Case, async: true
+  alias Yggdrasil.Player.Registry
+
+  setup do
+    {:ok, registry} = Registry.start_link
+    {:ok, registry: registry}
+  end
+
+  test "it allows a player to registry as online" do
+    user_id = 1
+    assert :ok = Registry.register_player user_id, self
+  end
+
+  test "it knows if a player is online" do
+    user_id = 1
+    refute Registry.is_online user_id
+    assert :ok = Registry.register_player user_id, self
+    assert Registry.is_online user_id
+  end
+
+  test "it will not allow the same player to register twice" do
+    user_id = 1
+
+    assert :ok = Registry.register_player user_id, self
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+    assert {:error, :already_registered} = Task.await(register_task)
+  end
+
+  test "it knows when a player exits" do
+    user_id = 1
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+
+    # verify player was registered successfully
+    assert :ok = Task.await(register_task)
+
+    refute Registry.is_online user_id
```

Comment:
Yeah this one fails because the lookup happens before the :DOWN message is handled. I'm thinking I'll add the ability to receive notifications when a player joins and leaves so I can use that to test this.